.github/workflows/python.yaml

```diff
@@ -0,0 +1,49 @@
+name: Python package
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [feature/capgen, main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.8', '3.9', '3.10', '3.11']
```

Comment:
Other CI tests go up to Python 3.12, is there a reason we can't also do that here?

Reply:
Good suggestion. I added 3.12 for the build tests, but ran into an annoying issue for the doctests and backed out of that - issue here: https://github.com/NCAR/ccpp-framework/issues/557

Reply:
Per our discussion yesterday, I changed the tests to no longer rely on the string representation of OrderedDict so now 3.12 is back in business!